test/extensions/filters/http/lua/lua_integration_test.cc

```diff
@@ -576,5 +638,178 @@ name: lua
+    response->waitForEndStream();
+
+    EXPECT_TRUE(response->complete());
+    EXPECT_EQ("200", response->headers().getStatusValue());
+  };
+
+  // Lua code defined in 'inline_code' will be executed by default.
+  Http::TestRequestHeaderMapImpl default_headers{{":method", "GET"},
+                                                 {":path", "/lua/per/route/default"},
+                                                 {":scheme", "http"},
+                                                 {":authority", "lua.per.route"},
+                                                 {"x-forwarded-for", "10.0.0.1"}};
+  check_request(default_headers, "code_from_global");
+
+  // Test whether LuaPerRoute can disable the Lua filter.
+  Http::TestRequestHeaderMapImpl disabled_headers{{":method", "GET"},
+                                                  {":path", "/lua/per/route/disabled"},
+                                                  {":scheme", "http"},
+                                                  {":authority", "lua.per.route"},
+                                                  {"x-forwarded-for", "10.0.0.1"}};
+  check_request(disabled_headers, "");
+
+  // Test whether LuaPerRoute can correctly reference Lua code defined in filter config.
+  Http::TestRequestHeaderMapImpl hello_headers{{":method", "GET"},
+                                               {":path", "/lua/per/route/hello"},
+                                               {":scheme", "http"},
+                                               {":authority", "lua.per.route"},
+                                               {"x-forwarded-for", "10.0.0.1"}};
+
+  check_request(hello_headers, "code_from_hello");
+
+  Http::TestRequestHeaderMapImpl byebye_headers{{":method", "GET"},
+                                                {":path", "/lua/per/route/byebye"},
+                                                {":scheme", "http"},
+                                                {":authority", "lua.per.route"},
+                                                {"x-forwarded-for", "10.0.0.1"}};
+  check_request(byebye_headers, "code_from_byebye");
+
+  // When the name referenced by LuaPerRoute does not exist, Lua filter does nothing.
+  Http::TestRequestHeaderMapImpl nocode_headers{{":method", "GET"},
+                                                {":path", "/lua/per/route/nocode"},
+                                                {":scheme", "http"},
+                                                {":authority", "lua.per.route"},
+                                                {"x-forwarded-for", "10.0.0.1"}};
+
+  check_request(nocode_headers, "");
+
+  // Update route config by RDS. Test whether RDS can work normally.
+  sendSotwDiscoveryResponse<envoy::config::route::v3::RouteConfiguration>(
+      Config::TypeUrl::get().RouteConfiguration,
+      {TestUtility::parseYaml<envoy::config::route::v3::RouteConfiguration>(UPDATE_ROUTE_CONFIG)},
```

Comment:
I think in here, we should try to change the source code name? So the `/lua/per/route/update` refers to `update.lua` at init, but then we need to change it to something else? (probably you can try to update `/lua/per/route/hello` to use the `update.lua` instead?).